server/monitor-types/real-browser-monitor-type.js

```diff
@@ -228,6 +228,22 @@ async function testRemoteBrowser(remoteBrowserURL) {
         throw new Error(e.message);
     }
 }
+
+/**
+ * Cleanup function to terminate all browser processes and clear cache after each monitoring check.
+ * @returns {Promise<void>}
+ */
+async function cleanupBrowser() {
+    if (browser) {
+        const contexts = browser.contexts();
+        for (const context of contexts) {
+            await context.clearCookies();
+            await context.clearPermissions();
+            await context.close();
+        }
+    }
+}
+
```

Comment:
For the current design, Uptime Kuma creates one browser instance for multiple monitors.
So I am afraid that your new fix is still not working, as closing all contexts will probably affect other monitors.

As I said before, the current code is working fine on Windows.
Uncommenting `headless: false,` can see behind the screens.

However, I don't have desktop environment right now on Linux, I can't debug it with `headless: false` on Linux

Reply:
_For the current design, Uptime Kuma creates one browser instance for multiple monitors.
So I am afraid that your new fix is still not working, as closing all contexts will probably affect other monitors._

that's what I suspected... I will send a fix now.

_As I said before, the current code is working fine on Windows.
Uncommenting headless: false, can see behind the screens._

As I posted in 
https://github.com/louislam/uptime-kuma/issues/3788 there are many chromium processes/tabs left opened after check() is called, and needs to be closed to prevent the DoS.